homeassistant/components/uvc/camera.py

```diff
@@ -106,6 +109,21 @@ def motion_detection_enabled(self):
         caminfo = self._nvr.get_camera(self._uuid)
         return caminfo['recordingSettings']['motionRecordEnabled']
 
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        caminfo = self._nvr.get_camera(self._uuid)
+        if caminfo['channels'][0]['isRtspEnabled']:
+            return SUPPORT_STREAM
+
+    @property
+    def stream_source(self):
+        """Return the stream source."""
+        caminfo = self._nvr.get_camera(self._uuid)
+        if caminfo['channels'][0]['isRtspEnabled']:
+            stream = caminfo['channels'][0]['rtspUris'][0].split(':')[2]
+            return 'rtsp://{}:{}'.format(self._addr, stream)
```

Comment:
I know I told you to pass in the IP address earlier, but this is the IP of the NVR.  Is the NVR the thing serving the RTSP stream?  See comment above regarding what comes back from `rtspUris`.

Looking at `camera_image` and `_login` methods, it appears the still image comes directly from the camera.  If `rtspUris` is not a fully qualified RTSP URI, but the feed is served from the camera, you might be able to use similar logic to login to the camera, and then use `self._connect_addr` as the host.

Reply:
Everything gets proxied through the NVR. And the data I get back is kind of weird. 

Say for example my NVR is running in Docker and lives at 10.0.1.2 and my camera is at 10.0.55.55, an example accessible RTSP stream would be `rtsp://10.0.55.55:7447/5c35744ae4b00e065c24865d_0` but the `rtspStreamUri` gets returned via the API as `rtsp://172.18.0.12:7447/5c35744ae4b00e065c24865d_0` (which is the address of the docker container the NVR lives in).